sql/analyzer/resolve_column_defaults.go

```diff
@@ -0,0 +1,488 @@
+	"st_multilinestringfromwkb": {},
+	"st_mpointfromtext": {},
+	"st_multipointfromtext": {},
+	"st_mpointfromwkb": {},
+	"st_multipointfromwkb": {},
+	"st_mpolyfromtext": {},
+	"st_multipolygonfromtext": {},
+	"st_mpolyfromwkb": {},
+	"st_multipolygonfromwkb": {},
+	"st_numgeometries": {},
+	"st_numinteriorring": {},
+	"st_numinteriorrings": {},
+	"st_numpoints": {},
+	"st_overlaps": {},
+	"st_pointfromgeohash": {},
+	"st_pointfromtext": {},
+	"st_pointfromwkb": {},
+	"st_pointn": {},
+	"st_polyfromtext": {},
+	"st_polygonfromtext": {},
+	"st_polyfromwkb": {},
+	"st_polygonfromwkb": {},
+	"st_simplify": {},
+	"st_srid": {},
+	"st_startpoint": {},
+	"st_swapxy": {},
+	"st_symdifference": {},
+	"st_touches": {},
+	"st_transform": {},
+	"st_union": {},
+	"st_validate": {},
+	"st_within": {},
+	"st_x": {},
+	"st_y": {},
+	"statement_digest": {},
+	"statement_digest_text": {},
+	"std": {},
+	"stddev": {},
+	"stddev_pop": {},
+	"stddev_samp": {},
+	"str_to_date": {},
+	"strcmp": {},
+	"subdate": {},
+	"substr": {},
+	"substring": {},
+	"substring_index": {},
+	"subtime": {},
+	"sum": {},
+	"sysdate": {},
+	"system_user": {},
+	"tan": {},
+	"time": {},
+	"time_format": {},
+	"time_to_sec": {},
+	"timediff": {},
+	"timestamp": {},
+	"timestampadd": {},
+	"timestampdiff": {},
+	"to_base64": {},
+	"to_days": {},
+	"to_seconds": {},
+	"trim": {},
+	"truncate": {},
+	"ucase": {},
+	"uncompress": {},
+	"uncompressed_length": {},
+	"unhex": {},
+	"unix_timestamp": {},
+	"updatexml": {},
+	"upper": {},
+	"user": {},
+	"utc_date": {},
+	"utc_time": {},
+	"utc_timestamp": {},
+	"uuid": {},
+	"uuid_short": {},
+	"uuid_to_bin": {},
+	"validate_password_strength": {},
+	"values": {},
+	"var_pop": {},
+	"var_samp": {},
+	"variance": {},
+	"version": {},
+	"wait_for_executed_gtid_set": {},
+	"wait_until_sql_thread_after_gtids": {},
+	"week": {},
+	"weekday": {},
+	"weekofyear": {},
+	"weight_string": {},
+	"year": {},
+	"yearweek": {},
+}
+
+func resolveColumnDefaults(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope) (sql.Node, error) {
+	span, _ := ctx.Span("resolveColumnDefaults")
+	defer span.Finish()
+
+	return plan.TransformUp(n, func(n sql.Node) (sql.Node, error) {
+		switch node := n.(type) {
+		case sql.SchemeModifiable:
+			sch := node.Schema()
+			var columns = make(map[string]indexedCol)
+			for i, col := range sch {
+				columns[strings.ToLower(col.Name)] = indexedCol{col, i}
+			}
+			newSch := make(sql.Schema, len(sch))
+
+			for i, col := range sch {
+				newCol := *col
+				if col.Default == nil || col.Default.Resolved() {
+					newSch[i] = &newCol
+					continue
+				}
+				if sql.IsTextBlob(newCol.Type) && newCol.Default.IsLiteral() {
+					return nil, fmt.Errorf("text/blob types may only have expression default values")
+				}
+				newDefaultExpr, err := expression.TransformUp(newCol.Default.Expression, func(e sql.Expression) (sql.Expression, error) {
+					switch expr := e.(type) {
+					case *expression.UnresolvedFunction:
```

Comment:
This should happen earlier, where the other functions are resolved. All functions should be resolved at this point, and here you're just doing validation on which functions are allowed in defaults.

A better way to accomplish this is to make the resolve_functions rule apply to the nodes you are interested in, and then make those nodes implement sql.Expressioner to get and set any column default value expressions. I think that should work, so give it a shot.